src/middlewared/middlewared/plugins/service.py

```diff
@@ -941,11 +941,12 @@ def _started_notify(self, verb, what):
         if systemdataset['syslog']:
             await self.restart("syslogd", kwargs)
         await self.restart("cifs", kwargs)
-        if systemdataset['rrd']:
-            # Restarting collectd may take a long time and there is no
-            # benefit in waiting for it since even if it fails it wont
-            # tell the user anything useful.
-            asyncio.ensure_future(self.restart("collectd", kwargs))
+
+        # Restarting rrdcached may take a long time and there is no
+        # benefit in waiting for it since even if it fails it wont
+        # tell the user anything useful.
+        asyncio.ensure_future(self.restart("rrdcached", kwargs))
```

Comment:
Same here, shouldnt we only restart collectd?

Reply:
No, here we need to restart rrdcached so it can manage files in new location/mount. Also we don't need to restart collectd because nothing has changed (has it? we don't monitor system dataset mount), just need to start it if it was in crashed state due to system dataset being absent previously.

Reply:
From what I see the location comes from collectd via unix socket. The location shouldnt matter even if we changed the system dataset, its always /var/db/collectd/rrd, it does a symlink.

The migration itself already takes care of restarting it, no?

Also it seems we are missing -F in rrdcached_flags?

```
SIGINT and SIGTERM

The daemon exits normally on receipt of either of these signals. Pending updates are handled in accordance with the -j and -F options.
```

Reply:
It seems not regarding -F, I think thats default if -j is not used.

Reply:
If rrdcached has some file open and we move that file/mount another FS/change sumlink, it won't know until we restart it.

Reply:
If thats moved wouldnt you already have done rrdcached stop+start on migrate?

Reply:
Great question, most likely legacy remaining of how system dataset was handled before middlewared plugin?

Reply:
Do you think we want to remove it altogether then?

Reply:
Only if you're confident enough it wont regress, otherwise we can keep it all.